Ginger/GingerCore/Actions/Rest/ActREST.cs

```diff
@@ -513,46 +515,87 @@ public override void Execute()
                                 {
                                     cks.Path = cks.Path.Substring(1);
                                 }
-                                try 
-                                { 
+                                try
+                                {
                                     _sessionCokkiesDic.Remove(cks.Name);
                                 }
-                                finally 
+                                finally
                                 {
                                     _sessionCokkiesDic.Add(cks.Name, cks);
                                 }
-                               
+
                             }
                         }
                     }
-                 
 
+
+                }
+
+                string resp = string.Empty;
+
+                if (RestResponseSave == true && ResponseContentType == eContentType.PDF)
+                {
+                    MemoryStream memoryStream = new MemoryStream();
+                    WebResponse webResponse = WebReq.GetResponse();
+
+                    webResponse.GetResponseStream().CopyTo(memoryStream);
+
+                    var headerkey = webResponse.Headers.AllKeys;
+                    var header = string.Join(",", headerkey);
+                    Reporter.ToLog(eAppReporterLogLevel.INFO, header);
+                    byte[] data = memoryStream.ToArray();
+
+
+                    String fileName = "";
+                    String timeStamp = DateTime.Now.ToString("dd_MM_yyyy_HH_mm_ss_fff");
+                    fileName += this.Description;
+                    fileName += "_" + "Response";
+                    fileName += "_" + timeStamp;
+
+                    string DirectoryPath = SaveRequestResponseFolderPath.ValueForDriver;
+                    if (DirectoryPath.StartsWith(@"~\"))
+                    {
+                        DirectoryPath = DirectoryPath.Replace(@"~\", SolutionFolder);
+                    }
+                    DirectoryPath += @"\" + "Response";
+
+                    if (!Directory.Exists(DirectoryPath))
+                    {
+                        Directory.CreateDirectory(DirectoryPath);
+                    }
+                    string filePath = DirectoryPath + "//" + fileName + ".pdf";
+
+                    File.WriteAllBytes(filePath, data);
+
+                }
+                else
+                {
+                    //TODO: check if UTF8 is good for all
```

Comment:
We always keep the happy path scenarios first. So move the else part to If and vice versa